pkg/apis/core/validation/validation.go

```diff
@@ -1262,6 +1271,21 @@ func validateRBDPersistentVolumeSource(rbd *core.RBDPersistentVolumeSource, fldP
 	if len(rbd.RBDImage) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("image"), ""))
 	}
+	if rbd.SecretRef != nil {
+		if len(rbd.SecretRef.Name) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "name"), ""))
+		} else {
+			for _, msg := range ValidateSecretName(rbd.SecretRef.Name, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("secretRef", "name"), rbd.SecretRef.Name, msg))
+			}
+		}
+		if len(rbd.SecretRef.Namespace) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "namespace"), ""))
+		} else {
+			allErrs = append(allErrs, ValidateDNS1123Label(rbd.SecretRef.Namespace, fldPath.Child("secretRef", "namespace"))...)
```

Comment:
It could be useful to declare `ValidateNamespaceName`

Reply:
This makes sense, Will add a function for the same

Reply:
ValidateNamespaceName function already exists.
[ValidateDNS1123Label](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L114) does the same thing as that of ValidateNamespaceName the only difference the return signature.
[ValidateDNS1123Label](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L114) return field.ErrorList and ValidateNamespaceName returns []string using which field.ErrorList has to be created.